addons/portal/wizard/portal_share.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+
+
+class PortalShare(models.TransientModel):
+    _name = 'portal.share'
+
+    @api.model
+    def default_get(self, fields):
+        result = super(PortalShare, self).default_get(fields)
+        result['res_model'] = self._context.get('active_model')
+        result['res_id'] = self._context.get('active_id')
+        result['share_link'] = self.env[result['res_model']].browse(result['res_id'])._get_share_url()
+        return result
+
+    res_model = fields.Char('Related Document Model', required=True)
+    res_id = fields.Integer('Related Document ID', required=True)
+    partner_ids = fields.Many2many('res.partner', string="Recipients", required=True)
+    note = fields.Text(help="Add extra content to display in the email")
+    share_link = fields.Char(string="Link", compute='_compute_share_link')
+    access_warning = fields.Text("Access warning", compute="_compute_access_warning")
+
+    @api.depends('res_model', 'res_id')
+    def _compute_share_link(self):
+        base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        for rec in self:
+            res_model = self.env[rec.res_model]
+            if isinstance(res_model, self.pool['portal.mixin']):
```

Comment:
I woud put an else: rec.share_link = False just to have a complete implementation (this should not occur but let us be complete)